infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LayoutGeometry.kt

```diff
@@ -155,8 +160,11 @@ interface IAlignment<M : AlignmentM<M>> : Loggable {
     fun getClosestPoint(target: IPoint, snapDistance: Double = 0.0): Pair<AlignmentPoint<M>, IntersectType>? =
         getClosestPointM(target)?.let { (m, type) -> getPointAtM(m, snapDistance)?.let { p -> p to type } }
 
-    fun getClosestPointM(target: IPoint): Pair<LineM<M>, IntersectType>? =
-        findClosestSegmentIndex(target)?.let { segmentIndex ->
+    fun getClosestPointM(
```

Comment:
Tällä oli ihan kohtuullisen tuntuva vaikutus. Rakenne on hiukan erikoinen, mutta koitin pitää vaikutuksen täsmällisenä vain siihen millä on oikeasti väliä: lähimmän pisteen haku pituusmittauslinjalta. Tai täsmällisemmin vielä tuosta algormitmista se osa joka hakee pistettä lähimmän segmentin. 

Alkuperäinen yleispätevä algoritmi toimii siis muilla viivatyypeillä edelleen mutta se on pituusmittauslinjalle yliajettu R-puu-haulla. R-puu alustetaan linjalle lazynä ensimmäisellä käyttökerralla. Tuon voisi ehkä tehdä initissäkin, mutta näin tehtynä oli selkeämpää todentaa suorituskykyvaikutus kun hintaa ei siirretä cachen lämmitykseen vaan se pysyy (ensimmäisellä) käyttöpaikalla.